src/linker/Linker.Steps/BodySubstituterStep.cs

```diff
@@ -9,45 +9,77 @@ namespace Mono.Linker.Steps
 {
 	public class BodySubstituterStep : BaseStep
 	{
-		protected override void Process ()
+		readonly XPathDocument _document;
+		readonly string _xmlDocumentLocation;
+		readonly string _resourceName;
+		readonly AssemblyDefinition _resourceAssembly;
+
+		public BodySubstituterStep (XPathDocument document, string xmlDocumentLocation = "<unspecified>")
```

Comment:
It's there so that we can report reasonable error messages for embedded XML files - see https://github.com/mono/linker/blob/17b3c016d78508b9f6c1f6a61a31b5786d91cf24/src/linker/Linker.Steps/ResolveFromXmlStep.cs#L96 and https://github.com/mono/linker/blob/69f31a90085e33dcde1b1ad900ed0c909bac578b/src/linker/Linker.Steps/BlacklistStep.cs#L110. If you're asking why it's optional - I don't see a good reason. I could make it required for both `BodySubstituterStep` and `ResolveFromXmlStep` if you like.

Reply:
> If you're asking why it's optional - I don't see a good reason.

Yep, I think it's not needed in this case